src/ui/components/auth/SignInContent.tsx

```diff
@@ -23,60 +24,84 @@ interface SignInContentProps {
 	onForgotPasswordBtnClick: () => void;
 }
 
+interface IErrorObject {
+	email: string;
+	password: string;
+}
+
 const SignInContent: React.FC<SignInContentProps> = ({
 	onSuccess,
 	onCreateNewUserClick,
 	onForgotPasswordBtnClick,
 }): JSX.Element => {
 	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [userTriedToSubmit, setUserTriedToSubmit] = useState(false);
 
 	const [email, setEmail] = useState("");
 	const [password, setPassword] = useState("");
-	const [emailError, setEmailError] = useState("");
-	const [passwordError, setPasswordError] = useState("");
 
-	const validateKey = async () => {
-		setIsLoading(true);
+	const [errors, setErrors] = useState<IErrorObject>({
+		email: "",
+		password: "",
+	});
+	const [serverValidationError, setServerValidationError] = useState("");
+
+	const validateCredentials = async () => {
 		const response = await fetchData({
-			url: getApiUrl("/api/key/validate"),
+			url: getApiUrl("/api/sample/validate"),
 			method: "POST",
 			config: {
 				headers: {
 					// authorization: `Bearer ${apiKey}`,
 				},
 			},
 		});
-
 		const body = await response.json();
-
-		if (response.status === 200 && body.status === "OK") {
-			// localStorageHandler.setItem(StorageKeys.API_KEY, apiKey);
-			onSuccess();
-		} else if (response.status === UNAUTHORISED_STATUS) {
-			// setApiKeyFieldError("Invalid API Key");
+		if (response.status === HTTPStatusCodes.OK) {
+			if (body.status === "OK") onSuccess();
+			else setServerValidationError("Incorrect email and password combination");
 		} else {
-			// setApiKeyFieldError("Something went wrong");
+			setServerValidationError("Something went wrong");
 		}
+	};
 
-		setIsLoading(false);
+	const checkValuesForErrors = () => {
+		const _errors = {
```

Comment:
Strongly type this to `IErrorObject` so that if the type changes in the future this will report errors